src/library/scala/StringContext.scala

```diff
@@ -117,6 +117,33 @@ case class StringContext(parts: String*) {
    */
   def raw(args: Any*): String = standardInterpolator(identity, args, parts)
 
+  /** Shorthand symbol constructor.
+   *
+   *  Here's an example of usage:
+   *  {{{
+   *    val abc = List(sym"a", sym"b", sym"c")
+   *    println(syms) // List('a, 'b, 'c)
+   *  }}}
+   *  This string-based shorthand, replaces the old built-in quote syntax:
+   *  {{{
+   *    val abc = List('a, 'b, 'c) // Quoted symbol literals dropped in 2.14
+   *  }}}
+   *
+   *  @param `str` The string to be converted to a symbol.
+   */
+  def sym(args: String*): Symbol = {
+    Symbol(standardInterpolator(processEscapes, args, parts))
+  }
+
+  object sym {
+    def unapplySeq(s: Symbol): Option[Seq[String]] = {
+      parts match {
+        case first +: Seq() if s.name == first => Some(Seq.empty[String])
+        case _                                 => None
+      }
```

Comment:
If not extracting values, `def unapply(s: Symbol): Boolean`, but maybe you intend to emulate the s-extractor. You can write `case s.name +: Seq() =>`.

Reply:
I guess it has to do with what the compiler is expecting for these string contexts?

If I try `def unapply(s: Symbol):Boolean`, 

```
[error] too many patterns for object sym offering Boolean: expected 0, found 1
[error]       case sym"$bar"          => assertTrue(false)
```

Reply:
I'm not planning on extracting values.  It looked difficult to make work.  I'm hoping it's enough of a corner case that it's not worth supporting, it wasn't supported before, and one that would be better served with String and an s-extractor.  

The problem is that any `$` in a symbol name the compiler will interpret as interpolation and be both troublesome and surprising for users when they are matching.

Reply:
There are similar gotchas for regex interpolation. Maybe throw an exception if `parts.length > 1`?

Reply:
Well, it would implicitly throw `MatchError` at runtime, but I agree that a more intentional exception would be better.  Added the following:
```scala
throw new IllegalArgumentException(
  "Variable interpolation not supported, use '$$' for '$'"
)
```